### ISSUE TITLE
false positive consider-using-namedtuple-or-dataclass

### ISSUE THREAD

**Issue by sam-s** [pylint-dev/pylint#10478]
### Bug description

```python
I use a dictionary of `matplotlib.Axes.plot` arguments in a dict:

line_params = {
    "regression":{
        "color": "green",
        "linewidth": 3,
        "linestyle": "dashed",
    },
    "diagonal":{
        "color": "red",
        "linewidth": 2,
    },
}

and use them like this:

ax.plot(..., **line_params[line_type])

the `line_params = {` line triggers

R6101: Consider using namedtuple or dataclass for dictionary values (consider-using-namedtuple-or-dataclass)

this is problematic for 2 reasons:
1. I do use the values of the `line_params` as a `dict`, so if I use, say, `dataclass`, I would need an extra `asdict` call.
2. the dictionarys have different key sets, so I would need to add an unnecessary field `"linestyle"` to `"diagonal"`
```

### Configuration

```ini

```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
R6101: Consider using namedtuple or dataclass for dictionary values (consider-using-namedtuple-or-dataclass)
```

### Expected behavior

no diagnostics

### Pylint version

```shell
pylint 3.3.7
astroid 3.3.11
Python 3.13.5 (tags/v3.13.5:6cb20a2, Jun 11 2025, 16:15:46) [MSC v.1943 64 bit (AMD64)]
```

### OS / Environment

MSYS_NT-10.0-19045

### Additional dependencies

```python

```
